### PR TITLE
snapcraft: nvidia-container: Enable shallow clone (5.0-candidate)

### PR DESCRIPTION
**Pull request by tomponline**
To reduce the size of the git repo clone (~2GB) which is causing failures on riscv64.

### CODE REVIEW

**Review by simondeziel — APPROVED** [review]

**Review by MusicDin — APPROVED** [review]

File: snapcraft.yaml [hunk 1]
```diff
@@ -582,10 +582,11 @@ parts:
       - libseccomp
     source: https://github.com/NVIDIA/libnvidia-container
     source-commit: 4c2494f16573b585788a42e9c7bee76ecd48c73d # v1.16.1
-    # XXX: fails to build if using source-commit and source-depth
-    #source-depth: 1
+    source-depth: 1
     source-type: git
     plugin: make
+    build-environment:
+      - GIT_TAG: "1.16.1" # Enables source-depth: 1, should match git tag without "v" prefix.
```
Comment (simondeziel) [inline snapcraft.yaml:589]: I wonder if it wouldn't be feasible to use `export GIT_TAG="$(git describe --tags --abbrev=0 | sed 's/^v//')"` in the override-build?

**Comment by tomponline** [comment]
I'll also apply this to our other series.

File: snapcraft.yaml (hunk 1 above)
Comment (tomponline) [reply snapcraft.yaml:589]: I initially tried doing `export GIT_TAG="1.16.1"` and it didnt work when using `snapcraftctl build` so looks like it resets the env

Comment (simondeziel) [reply snapcraft.yaml:589]: That's why I suggested having that done in the `override-build` section. Unless you've had that export put there already?

Comment (tomponline) [reply snapcraft.yaml:589]: yeah thats right it was there

Comment (tomponline) [reply snapcraft.yaml:589]: before the call to `snapcraftctl build`

**Pull request merged**